### ISSUE TITLE
Cheat Repositories

### ISSUE THREAD

**Issue by senaps** [MCSH/cheat#2]
so, thing is, I always wanted to re-do a thing that I have done before but can't remember how I did it! searching is the answer but is time-consuming. we forget how we did something all the time right?

this project allows me to fix that problem, but the thing is I need to move the cheat files all over the places I want to use'em.
The idea is to make it a `.com` thing so people can share their cheats. maybe even have their own local cheat files too. or remote repo for their not public cheat files. it's quickly turning into some multi repo thing!:) where we have an official repo, and then each user may or may not have their own shared repo across their network, and then their local repo's. (choosing what repo's to include in the app should be up to the user right?)

Like I hate it when a co-workers swim's by to ask me `how would I create a decorator?` I can point him to our office's cheats repository and he is good since I have written the cheat on writing a decorator, and it's following the exact policy of our company. 

currently, I have no idea on how to do it or where to start, and I didn't like the idea of using `YAML` files.
we may need a way to quickly add cheat things too.

Let's collaborate on it a bit and see what it grows to be.

راستی میشه اینجا راحت فارسی نوشت؟
ادیت۲: میشه ولی راست به چپش مشکل داره که من احتمالا باهاش اوکی باشم! من یکی که سویچ کردم رو فارسی واسه جواب دادن مگر بخوایم ریپو تر تمیز بمونه که شاید استفاده غیر فارسی زبان هم داشت!

**Comment by MCSH** [comment]
Here is the thing, cheat supports multiple cheat directories through the variable `CHEATPATH`. So here the schema I would suggest for your said scenario:

The user will create multiple folders to store cheats. Some are, of course private, and some are shared. The shared ones could use `git` as a method of synchronizing across devices. So we do need a way to keep these directories up to date. I would suggest either another flag or a new command, such as `cheatrepo` that would do these things. Any ideas?

---
Regarding YAML files, there are to options currently, either running the command `cheat --add ....` or editing the YAML files. The choice of YAML files was so that both users and machine could read/parse the file just by looking. 

---
من ترجیح میدم فارسی نباشه چون کاربر غیر ایرانی داره، اگه نیاز به گفت‌و‌گوی فارسی هست میتونیم توییتر یا تلگرام صحبت کنیم درباره‌اش :))

**Comment by ahangarha** [comment]
<div dir=rtl>
برای فارسی نویسی در گیت‌هاب باید قبل از متن فارسی این رو قرار بدید
</div>

    < div dir=rtl>

<div dir=rtl>
البته برای اینکه گیت‌هاب بتونه کد رو نشون بده، یک فاصله اضافه زده‌ام قبل از div
</div>

**Comment by ahangarha** [comment]
Though I am getting the sense of this project but it is still ambiguous how it is being used.

**Comment by ahangarha** [comment]
Is this a right way to use this cheat?

```
(env) ➜  cheat git:(sandbox) ✗ ./cheat.py vim -t paste
To be able to paste some text in regular way inside VIM, you need to type:

:set pastetoggle

Then enter INSERT mode and paste the text using CTRL+SHIFT+V

(env) ➜  cheat git:(sandbox) ✗ 
```

**Comment by senaps** [comment]
agreed, documentation on how to use it isn't that clear, I'll try and update the documents too if @MCSH didn't do so.

**Comment by ahangarha** [comment]
I am making one cheat sheet for myself based on the what I have sent above. It is for what I do with find command.

If the way I use it be right, I have some suggestion on how to make the command easier to use.

Meanwhile, I hope @MCSH can share what he does (or has in mind to do) with this program. Even one short video would be nice for introducing the project. All depends on him. :)

**Comment by MCSH** [comment]
I'll try to record something. Although suggesting a are welcome!

Although how you used it is almost what I had in mind!
